net/src/ipc_net_worker.rs

```diff
@@ -126,60 +119,34 @@ impl IpcNetWorker {
         ipc_uri: String,
         done: NetShutdown,
         bootstrap_nodes: Vec<String>,
-        endpoint: String,
     ) -> NetResult<Self> {
         let log = TweetProxy::new("IpcNetWorker");
         log.i(&format!("connect to uri {}", ipc_uri));
 
-        let mut socket = TransportWss::with_std_tcp_stream();
-        wait_connect(&mut socket, &ipc_uri)?;
+        let mut wss_socket = TransportWss::with_std_tcp_stream();
+        let transport_id = wss_socket.wait_connect(&ipc_uri)?;
 
-        log.i("connection success");
+        log.i(&format!("connection success. tId = {}", transport_id));
 
         Ok(IpcNetWorker {
             handler,
-            socket,
+            wss_socket,
             ipc_uri,
+            transport_id,
             done,
-            is_ready: false,
+            is_network_ready: false,
             last_known_state: "undefined".to_string(),
             last_state_millis: 0.0_f64,
             bootstrap_nodes,
-            endpoint,
             log,
         })
     }
 }
 
-fn wait_connect(
-    socket: &mut TransportWss<std::net::TcpStream>,
-    uri: &str,
-) -> NetResult<Vec<TransportEvent>> {
-    let mut out = Vec::new();
-
-    socket.connect(&uri)?;
-
-    let start = std::time::Instant::now();
-    while (start.elapsed().as_millis() as usize) < DEFAULT_HEARTBEAT_WAIT_MS {
-        let (_did_work, evt_lst) = socket.poll()?;
-        for evt in evt_lst {
-            match evt {
-                TransportEvent::Connect(_id) => {
-                    return Ok(out);
-                }
-                _ => out.push(evt),
-            }
-        }
-        std::thread::sleep(std::time::Duration::from_millis(3));
-    }
-
-    bail!("ipc connection timeout");
-}
-
 impl NetWorker for IpcNetWorker {
     /// stop the net worker
     fn stop(mut self: Box<Self>) -> NetResult<()> {
-        self.socket.close_all()?;
+        self.wss_socket.close(self.transport_id)?;
```

Comment:
while there isn't currently a path that creates multiple connections, shouldn't we still close them all just in case?

Reply:
Yeah. There is currently an issue with closing with n3h-mock and I thought this would fix it but it actually doesn't so i can revert.

Reply:
reverted